gxdynamodb/src/main/java/com/genexus/db/dynamodb/DynamoDBConnection.java

```diff
@@ -0,0 +1,145 @@
+package com.genexus.db.dynamodb;
+
+import com.genexus.db.service.ServiceConnection;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClientBuilder;
+
+import java.net.URI;
+import java.sql.ResultSet;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+
+public class DynamoDBConnection extends ServiceConnection
+{
+	private static final String GXDYNAMODB_VERSION = "1.0";
+
+	private static final String CLIENT_ID = "user";
+	private static final String CLIENT_SECRET = "password";
+	private static final String REGION = "region";
+	private static final String LOCAL_URL = "localurl";
+
+
+	DynamoDbClient mDynamoDB;
+	Region mRegion = Region.US_EAST_1;
+
+	public DynamoDBConnection(String connUrl, Properties initialConnProps)
+	{
+		super(connUrl, initialConnProps); // After initialization use props variable from super class to manage properties
+		initializeDBConnection(connUrl);
+	}
+
+	private void initializeDBConnection(String connUrl)
+	{
+		String mLocalUrl = null, mClientId = null, mClientSecret = null;
+		for(Enumeration<Object> keys = props.keys(); keys.hasMoreElements(); )
+		{
+			String key = (String)keys.nextElement();
+			String value = props.getProperty(key, key);
+			switch(key.toLowerCase())
+			{
+				case LOCAL_URL: mLocalUrl = value; break;
+				case CLIENT_ID: mClientId = value; break;
+				case CLIENT_SECRET: mClientSecret = value; break;
+				case REGION: mRegion = Region.of(value); break;
+				default: break;
+			}
+		}
+		DynamoDbClientBuilder builder = DynamoDbClient.builder().region(mRegion);
+		if(mLocalUrl != null)
+			builder = builder.endpointOverride(URI.create(mLocalUrl));
+		if(mClientId != null && !mClientId.equals("") &&
```

Comment:
Maybe you can use isEmpty instead of equals or StringUtils.isNotEmpty to avoid ask for null